packages/wrangler/src/deploy/deploy.ts

```diff
@@ -588,7 +592,15 @@ See https://developers.cloudflare.com/workers/platform/compatibility-dates for m
 				})
 			: undefined;
 
-		const assets = await syncAssets(
+		// Upload assets if experimental assets is being used
+		const experimentalAssetsJwt = await syncExperimentalAssets(
+			accountId,
+			scriptName,
+			props.experimentalAssets?.directory,
+			props.dryRun
+		);
```

Comment:
Can we explicitly gate this behind the experimental_assets flag, to avoid running any new code at all when the flag is not enabled.